commands/device/stop-application.ts

```diff
@@ -0,0 +1,36 @@
+
+///<reference path="../../../.d.ts"/>
+"use strict";
+
+import helpers = require("./../../helpers");
+
+export class StopApplicationOnDeviceCommand implements ICommand {
+
+	constructor(private $devicesServices: Mobile.IDevicesServices,
+		private $errors: IErrors,
+		private $stringParameter: ICommandParameter,
+		private $staticConfig: Config.IStaticConfig,
+		private $options: IOptions) { }
+
+	allowedParameters: ICommandParameter[] = [this.$stringParameter];
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			this.$devicesServices.initialize({ deviceId: this.$options.device, skipInferPlatform: true }).wait();
+
+			if (this.$devicesServices.deviceCount > 1) {
+				this.$errors.fail("More than one device found. Specify device explicitly with --device option.To discover device ID, use $%s device command.", this.$staticConfig.CLIENT_NAME.toLowerCase());
```

Comment:
Shouldn't it be `failWithoutHelp`, the message gives enough information how to fix the problem